cmd/release-tools/github.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"golang.org/x/sys/unix"
+)
+
+type GitHubClient struct {
+	owner string
+	github.Client
+}
+
+func newGitHubClient(owner string, token string) (*GitHubClient, error) {
+	ctx := context.Background()
+
+	var tc *http.Client
+
+	if token != "" {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		)
+		tc = oauth2.NewClient(ctx, ts)
+	}
+
+	return &GitHubClient{owner, *github.NewClient(tc)}, nil
+}
+
+func nextBump(v string) (semver.Version, error) {
+	versionBump, err := semver.Make(v)
+	if err != nil {
+		return versionBump, err
+	}
+	versionBump.Patch++
+	return versionBump, nil
+}
+
+func checkAssets(assets []string) error {
+	for _, asset := range assets {
+		fmt.Println("Checking file", asset)
+		st, err := os.Stat(asset)
+		if err != nil {
+			return err
+		}
+
+		mode := st.Mode()
+		fmt.Printf("is a regular file... ")
+		if !mode.IsRegular() {
+			fmt.Println("FAIL")
+			return fmt.Errorf("Not a regular file")
+		}
+		fmt.Println("OK")
+
+		fmt.Printf("has read permissions ...")
+		if unix.Access(asset, unix.W_OK) != nil {
+			fmt.Println("FAIL")
+			return err
+		}
+		fmt.Println("OK")
+	}
+	return nil
+}
+func (c *GitHubClient) uploadAsset(repo string, releaseID int, asset string) error {
+
+	timeoutShortRequest := 5 * time.Second
```

Comment:
`timeoutShortRequest` is defined multiple times, should it be a constant/global var?